### PR TITLE
sign up flow

### PR DESCRIPTION
**Pull request by whatasunnyday**
Hey @siruguri,

Here is a list of my TODOs for commit but I think this PR is ready for a review. Most of them are visual. The only bit of "real work" is sending emails and I'm not sure if you want to block the request to send emails (like it is currently) or use activejob.

Fix visual errors on forms on account creation and payments
update copy on ""
update image on ""
prettify client errors (displayed by Stripe.js, see payments.js.coffee and span#payment-errors.
prettify server errors (displayed by flash[:alert])
coupon button styling (add welcome message)
check mailer to make sure it makes sense
after gift purchase page or nice message

add some feature specs
  -sign in
    with/without payment
    with/without onboarding
  - sign up (so far, controller + model tests are good)
  - buy gift
    signed in
    signed out

ACTUAL WORK: send email after gift purchase

unit test payment model?
unit test user model?

set up CI server
set up github branch deploy

### CODE REVIEW

**Comment by siruguri** [comment]
Looks good - I'd say don't worry about the ActiveJob transformation for
mailing.

File: app/controllers/payments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    @gift = Gift.new(gift_params)
+    @gift.build_coupon
+    @payment = Payment.new(payment_params)
+    @payment.email = @gift.your_email
+    @payment.coupon = found_coupon?
+    @payment.gift = true
+    if @gift.valid? && @payment.valid?
+      ActiveRecord::Base.transaction do
+        @gift.save
+        @payment.save(validate: false)
```
Comment (whatasunnyday) [inline app/controllers/payments_controller.rb]: really need to refactor this. the issue is that validate calls stripe for a token. save and valid? both call validations. perhaps i can memoize the value.

**Comment by whatasunnyday** [comment]
- [ ] Fix visual errors on forms on account creation and payments
- [ ] update copy on ""
- [ ] update image on ""
- [ ] prettify client errors (displayed by Stripe.js, see payments.js.coffee and span#payment-errors.
- [ ] prettify server errors (displayed by flash[:alert])
- [ ] coupon button styling (add welcome message)
- [ ] check mailer to make sure it makes sense
- [ ] after gift purchase page or nice message

- [ ] add some feature specs
- [ ] -sign in 
   - [ ] with/without payment
   - [ ] with/without onboarding

- [ ] sign up (so far, controller + model tests are good)
  - [ ] buy gift signed in signed out

- [ ] ACTUAL WORK: send email after gift purchase

- [ ] unit test payment model?
- [ ] unit test user model?
- [ ] unit test gift model?

- [ ] set up CI server
- [ ] set up github branch deploy

File: app/controllers/payments_controller.rb (hunk 1 above)
Comment (siruguri) [inline app/controllers/payments_controller.rb]: Yeah, wow, possibly the worst decision made by Julian during that weekend :)

**Pull request closed without merging**